tests/unittests/detection/test_panoptic_quality.py

```diff
@@ -26,113 +27,143 @@
-            ],
+            [[6, 0], [0, 0], [6, 0], [6, 0], [0, 1]],
+            [[0, 0], [0, 0], [6, 0], [0, 1], [0, 1]],
+            [[0, 0], [0, 0], [6, 0], [0, 1], [1, 0]],
+            [[0, 0], [7, 0], [6, 0], [1, 0], [1, 0]],
+            [[0, 0], [7, 0], [7, 0], [7, 0], [7, 0]],
         ]
-    ),
+    )
+    .reshape((1, 1, 5, 5, 2))
+    .repeat(2, 1, 1, 1, 1),
     target=torch.tensor(
         [
-            [
-                [[6, 0], [6, 0], [6, 0], [6, 0], [0, 0]],
-                [[0, 1], [0, 1], [6, 0], [0, 0], [0, 0]],
-                [[0, 1], [0, 1], [6, 0], [1, 0], [1, 0]],
-                [[0, 1], [7, 0], [7, 0], [1, 0], [1, 0]],
-                [[0, 1], [7, 0], [7, 0], [7, 0], [7, 0]],
-            ],
-            [
-                [[6, 0], [6, 0], [6, 0], [6, 0], [0, 0]],
-                [[0, 1], [0, 1], [6, 0], [0, 0], [0, 0]],
-                [[0, 1], [0, 1], [6, 0], [1, 0], [1, 0]],
-                [[0, 1], [7, 0], [7, 0], [1, 0], [1, 0]],
-                [[0, 1], [7, 0], [7, 0], [7, 0], [7, 0]],
-            ],
+            [[6, 0], [6, 0], [6, 0], [6, 0], [0, 0]],
+            [[0, 1], [0, 1], [6, 0], [0, 0], [0, 0]],
+            [[0, 1], [0, 1], [6, 0], [1, 0], [1, 0]],
+            [[0, 1], [7, 0], [7, 0], [1, 0], [1, 0]],
+            [[0, 1], [7, 0], [7, 0], [7, 0], [7, 0]],
         ]
-    ),
+    )
+    .reshape((1, 1, 5, 5, 2))
+    .repeat(2, 1, 1, 1, 1),
 )
-_args = {"things": {0, 1}, "stuffs": {6, 7}}
+_INPUTS_1: Final = Input(
+    # Shape of input tensors is (num_batches, batch_size, num_points, 2).
+    preds=torch.tensor(
+        [[10, 0], [10, 123], [0, 1], [10, 0], [1, 2]],
+    )
+    .reshape((1, 1, 5, 2))
+    .repeat(2, 1, 1, 1),
+    target=torch.tensor(
+        [[10, 0], [10, 0], [0, 0], [0, 1], [1, 0]],
+    )
+    .reshape((1, 1, 5, 2))
+    .repeat(2, 1, 1, 1),
+)
+_ARGS_0: Final = {"things": {0, 1}, "stuffs": {6, 7}}
+_ARGS_1: Final = {"things": {2}, "stuffs": {3}, "allow_unknown_preds_category": True}
+_ARGS_2: Final = {"things": {0, 1}, "stuffs": {10, 11}}
 
+# TODO: Improve _compare_fn by calling https://github.com/cocodataset/panopticapi/blob/master/panopticapi/evaluation.py
+# directly and compare at runtime on multiple examples.
 
-def _compare_fn(preds, target) -> np.ndarray:
-    """Reference implementation
 
-    Improve this by calling https://github.com/cocodataset/panopticapi/blob/master/panopticapi/evaluation.py
-    directly and compare at runtime on multiple examples
-    """
+def _compare_fn_0_0(preds, target) -> np.ndarray:
+    """Reference result for the _INPUTS_0, _ARGS_0 combination."""
     return np.array([0.7753])
 
 
+def _compare_fn_0_1(preds, target) -> np.ndarray:
+    """Reference result for the _INPUTS_0, _ARGS_1 combination."""
+    return np.array([np.nan])
+
+
+def _compare_fn_1_2(preds, target) -> np.ndarray:
+    """Reference result for the _INPUTS_1, _ARGS_2 combination."""
+    return np.array([(2 / 3 + 1 + 2 / 3) / 3])
```

Comment:
These are kind of ugly, but lambda functions aren't allowed because of multiprocessing and a lookup table doesn't seem to make sense either when the signature of this function requires two tensors and only that

Reply:
that is fine :)